cmd/admin-router.go

```diff
@@ -36,6 +37,26 @@ const (
 // adminAPIHandlers provides HTTP handlers for MinIO admin API.
 type adminAPIHandlers struct{}
 
+// adminApiHostHandler - allow access to the  Admin APIs only from local interface by default.
+func adminApiHostHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if globalIsGateway && globalGatewayName == PANFSBackendGateway && globalPanOnlyLocalAdminApi {
```

Comment:
Maybe it makes sense to apply this midlleware only when the server is running on the `PANFSBackendGateway` mode? Then you can remove one `if` from this function. It seems to be more efficient than checking for the server mode for each request.

Reply:
updated